src/coprocessor/mod.rs

```diff
@@ -159,7 +159,11 @@ impl ReqContext {
         cache_match_version: Option<u64>,
         perf_level: PerfLevel,
     ) -> Self {
-        let deadline = Deadline::from_now(max_handle_duration);
+        let mut deadline_duration = max_handle_duration;
+        if context.max_execution_duration_ms > 0 {
```

Comment:
should needs to check the `max_execution_duration_ms` and `deadline_duration`?

Reply:
I plan to let max_execution_duration_ms has higher priority